rustls/src/msgs/handshake.rs

```diff
@@ -968,18 +968,14 @@ impl ClientHelloPayload {
 
     pub(crate) fn has_keyshare_extension_with_duplicates(&self) -> bool {
         if let Some(entries) = self.keyshare_extension() {
-            let mut seen = BTreeSet::new();
-
-            for kse in entries {
-                let grp = u16::from(kse.group);
-
-                if !seen.insert(grp) {
-                    return true;
-                }
-            }
+            has_duplicates::<_, _, u16>(
+                entries
+                    .iter()
+                    .map(|kse| u16::from(kse.group)),
+            )
+        } else {
+            false
```

Comment:
nit: I think this could be a separate commit since it's unrelated to cert compression.